### PR TITLE
Enhance unit test coverage for go sensor

### PR DESCRIPTION
**Pull request by Angith**

### CODE REVIEW

File: integration_test.go [hunk 1]
```diff
@@ -60,6 +60,12 @@ func (srv *serverlessAgent) HandleBundle(w http.ResponseWriter, req *http.Reques
 		body = nil
 	}
 
+	if value, ok := os.LookupEnv("NEED_ERROR"); ok && value == "true" {
```
Comment (sanojsubran) [inline integration_test.go:63]: Query: Is it possible to remove env wide setting?

File: lambda_agent.go [hunk 2]
```diff
@@ -163,8 +163,9 @@ func (a *lambdaAgent) sendRequest(req *http.Request) error {
 			return nil
 		}
 
-		a.logger.Info("serverless agent has responded with ", resp.Status, ": ", string(respBody))
-		return nil
+		err = fmt.Errorf("serverless agent has responded with %s : %s", resp.Status, string(respBody))
```
Comment (sanojsubran) [inline lambda_agent.go:166]: Suggestion: Better to move to another PR as it has functional implications

**Review by sanojsubran — COMMENTED** [review]
Please have a look at the comments

File: instrumentation/instabeego/web_server_handler_test.go [hunk 3]
```diff
@@ -64,31 +63,31 @@ func initBeeApp(t *testing.T, r *instana.Recorder) {
 	sleep(serverDepTime)
 }
 
-func TestMain(m *testing.M) {
-
-	instana.InitSensor(&instana.Options{
+func initSensor() {
+	instana.StartMetrics(&instana.Options{
```
[the remark this answers is not in the document]
Comment (Angith) [reply instrumentation/instabeego/web_server_handler_test.go]: yes. Done 👍

File: integration_test.go (hunk 1 above)
Comment (Angith) [reply integration_test.go]: changed the settings into a tag in spans.

File: lambda_agent.go (hunk 2 above)
Comment (Angith) [reply lambda_agent.go]: done 👍

File: integration_test.go [hunk 4]
```diff
@@ -60,6 +61,20 @@ func (srv *serverlessAgent) HandleBundle(w http.ResponseWriter, req *http.Reques
...
+		log.Printf("ERROR: failed to unmarshal serverless agent spans request body: %s", err.Error())
+	} else {
+		if len(root.Spans) > 0 {
+			if root.Spans[0].Data.SDKCustom.Tags.ReturnError == "true" ||
+				root.Spans[0].Data.Lambda.ReturnError == "true" {
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+		}
+	}
```
Comment (sanojsubran) [inline integration_test.go:76]: Suggestion: Just combined the if checks.
```suggestion
if err != nil {
		log.Printf("ERROR: failed to unmarshal serverless agent spans request body: %s", err.Error())
	} else {
		if len(root.Spans) > 0 && (root.Spans[0].Data.SDKCustom.Tags.ReturnError == "true" ||
			root.Spans[0].Data.Lambda.ReturnError == "true") {

			w.WriteHeader(http.StatusInternalServerError)
			return
		}
	}
```

File: instrumentation/instabeego/web_server_handler_test.go [hunk 5]
```diff
@@ -64,31 +63,31 @@ func initBeeApp(t *testing.T, r *instana.Recorder) {
 	sleep(serverDepTime)
 }
 
-func TestMain(m *testing.M) {
-
-	instana.InitSensor(&instana.Options{
+func initSensor() {
```
Comment (sanojsubran) [inline instrumentation/instabeego/web_server_handler_test.go:66]: Can we change this to `initCollector` to remove the ambiguity ?

**Review by sanojsubran — APPROVED** [review]
Looks good. Please check the comments.

File: instrumentation/instabeego/web_server_handler_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sanojsubran) [reply instrumentation/instabeego/web_server_handler_test.go]: It would be better to rename the parent function also. We do not want to further advertise the term `sensor` in our `tracer` code.

[the remark this answers is not in the document]
Comment (Angith) [reply instrumentation/instabeego/web_server_handler_test.go]: make sense. Updated

File: instrumentation/instabeego/web_server_handler_test.go (hunk 5 above)
Comment (Angith) [reply instrumentation/instabeego/web_server_handler_test.go]: yes 👍
